src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -1236,6 +1409,29 @@ public IEnumerable<string> GetCertificateKeysFromLocalState(TransactionOperation
             }
         }
 
+        public IEnumerable<(string Thumbprint, BlittableJsonReaderObject Certificate)> GetAllCertificatesFromCluster(TransactionOperationContext context, int start, int take)
+        {
+            var certTable = context.Transaction.InnerTransaction.OpenTable(CertificatesSchema, CertificatesSlice);
+            
+            foreach (var result in certTable.SeekByPrimaryKeyPrefix(Slices.Empty, Slices.Empty, start))
+            {
+                if (take-- <= 0)
+                    yield break;
+
+                yield return GetCurrentItem(context, result.Value);
+            }
+        }
+
+        public IEnumerable<string> GetCertificateKeysFromCluster(TransactionOperationContext context)
```

Comment:
What is this method used for?

Reply:
Renamed it to GetCertificateThumbprintsFromCluster. It is used to get all the keys so that we can delete them all from local state in OnSnapshotInstalledAsync() and in DeleteAllExistingCertificates()